src/content/en/updates/2018/06/audio-worklet-design-pattern.md

```diff
@@ -0,0 +1,414 @@
+{# wf_featured_snippet: Advanced design patterns to unlock AudioWorklet's fullest power with WebAssembly and SharedArrayBuffer. #}
+{# wf_blink_components: Blink>WebAudio #}
+
+<style>
+  .diagram { max-width: 90%; display: block; margin: 0 auto; padding: 12px; }
+  figcaption { font-size: 14px; text-align: center; padding: 8px; }
+</style>
+
+# AudioWorklet Design Pattern {: .page-title }
+
+{% include "web/_shared/contributors/hongchanchoi.html" %}
+
+The [previous article](/web/updates/2017/12/audio-worklet)
+on the AudioWorklet detailed the basic concepts and its usage. Since its launch
+in Chrome 66 there have been many requests for more examples of how it can be
+used in actual applications. The AudioWorklet unlocks the full potential of
+WebAudio, but taking advantage of it can be challenging because it requires
+understanding concurrent programming wrapped with several JS APIs. Even for
+developers who are familiar with the WebAudio, integrating the AudioWorklet
+with other APIs (e.g. WebAssembly) can be difficult.
+
+This article will give the reader a better understanding of how to use the
+AudioWorklet in real-world settings and to offer tips to draw on its fullest
+power. Be sure to check out
+[code examples and live demos](https://googlechromelabs.github.io/web-audio-samples/audio-worklet#design-pattern)
+as well!
+
+## Recap: AudioWorklet
+
+Before diving in, let's quickly recap terms and facts around the AudioWorklet
+system which is previously introduced in
+[this post](/web/updates/2017/12/audio-worklet).
+
+  - [BaseAudioContext](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext):
+    Web Audio API's primary object.
+  - [AudioWorklet](https://webaudio.github.io/web-audio-api/#audioworklet): A
+    special script file loader for the AudioWorklet operation. Belongs to
+    BaseAudioContext. A BaseAudioContext can have one AW. The loaded script file
+    is evaluated in the AudioWorkletGlobalScope and is used to create the
+    AudioWorkletProcessor instances.
+  - [AudioWorkletGlobalScope](https://webaudio.github.io/web-audio-api/#audioworkletglobalscope)
+    : A special JS global scope for the AudioWorklet operation. Runs on a
+    dedicated rendering thread for the WebAudio. A BaseAudioContext can have one
+    AudioWorkletGlobalScope.
+  - [AudioWorkletNode](https://webaudio.github.io/web-audio-api/#audioworkletnode)
+    : An AudioNode designed for the AudioWorklet operation. Instantiated from a
+    BaseAudioContext. A BaseAudioContext can have multiple AudioWorkletNodes
+    similarly to the native AudioNodes.
+  - [AudioWorkletProcessor](https://webaudio.github.io/web-audio-api/#audioworkletprocessor)
+    : A counterpart of the AudioWorkletNode. The actual guts of the
+    AudioWorkletNode processing the audio stream by the user-supplied code. It
+    is instantiated in the AudioWorkletGlobalScope when a AudioWorkletNode is
+    constructed. An AudioWorkletNode can have one matching
```

Comment:
"can have one" -> "can have only one".  But I guess it's impossible to have more than one, so I guess this is ok.